File: src/18p1.py
with open("../inp/18.txt", 'r') as f:
    lines = f.readlines()

r = 0
c = 0
points = [(r, c)]
boundary_points = 0
for line in lines:
    d, cnt = line.split(" ")[:2]
    if d == "R":
        c += int(cnt)
    elif d == "L":
        c -= int(cnt)
    elif d == "D":
        r += int(cnt)
    else:
        r -= int(cnt)
    points.append((r, c))
    boundary_points += int(cnt)

area = 0
for (p1, p0) in zip(points[1:], points[:-1]):
    area += (p0[0] + p1[0]) * (p0[1] - p1[1])
area = abs(area) // 2

interior_points = area - boundary_points // 2 + 1
print(interior_points + boundary_points)

File: src/18p1.kt
import kotlin.math.abs

fun main() {
    val lines = readInput("18")

    data class Point(val r: Int, val c: Int)

    var r = 0
    var c = 0
    var cntBoundaryPoints = 0
    val points = mutableListOf<Point>()

    lines.forEach {
        val (d, cntStr) = it.split(" ").take(2)
        val cnt = cntStr.toInt()
        when (d) {
            "R" -> c += cnt
            "L" -> c -= cnt
            "D" -> r += cnt
            "U" -> r -= cnt
            else -> throw Exception("Unexpected direction $d")
        }
        points.add(Point(r, c))
        cntBoundaryPoints += cnt
    }

    val area = abs(points.zipWithNext().sumOf { (it.first.r + it.second.r) * (it.first.c - it.second.c) }) / 2
    val cntInteriorPoints = area - cntBoundaryPoints / 2 + 1
    (cntBoundaryPoints + cntInteriorPoints).println()
}